app/charts/scatterplot/scatterplot-state.tsx

```diff
@@ -72,15 +72,14 @@ const useScatterplotState = ({
   const getY = useOptionalNumericVariable(fields.y.componentIri);
   const getSegment = useSegment(fields.segment?.componentIri);
 
-  // All data, sort by segment
-  const sortedData = data.sort((a, b) =>
-    ascending(getSegment(a), getSegment(b))
-  );
+  const plottableSortedData = data
+    .filter((d) => getX(d) !== null && getY(d) !== null)
+    .sort((a, b) => ascending(getSegment(a), getSegment(b)));
```

Comment:
Can use useMemo here ? Since we are dealing with a potentially big array + a sort :)

Reply:
Actually I see that this could be done in usePreparedData. I think it makes sense to have it as a default filter, and it'd be memoed directly.

Reply:
I tried this and realised that then there are situations where we have more items in ColorLegend below the chart than visible on the chart itself, and it shouldn't be like that (as we only filter preparedData, while segments are created basing on "full" data). I refactored the code to have a function that does this filtering across all charts. I think it would make sense to test it out, as I am not sure if it won't break some time filtering stuff 😆